src/crypto/bn.js

```diff
@@ -1,130 +1,156 @@
-'use strict';
 
-var BN = require('bn.js');
-var $ = require('../util/preconditions');
-var _ = require('lodash');
 
-var reversebuf = function(buf) {
-  var buf2 = new Buffer(buf.length);
-  for (var i = 0; i < buf.length; i++) {
+const BN = require('bn.js');
+const _ = require('lodash');
+const $ = require('../util/preconditions');
+
+function reversebuf(buf) {
+  /* eslint-disable */
+  const buf2 = Buffer.alloc(buf.length);
+    /* eslint-enable */
+  for (let i = 0; i < buf.length; i += 1) {
     buf2[i] = buf[buf.length - 1 - i];
   }
   return buf2;
-};
+}
 
 BN.Zero = new BN(0);
 BN.One = new BN(1);
 BN.Minus1 = new BN(-1);
 
-BN.fromNumber = function(n) {
+
+function fromNumber(n) {
   $.checkArgument(_.isNumber(n));
   return new BN(n);
-};
+}
+
+BN.fromNumber = fromNumber;
 
-BN.fromString = function(str, base) {
+function fromString(str, base) {
   $.checkArgument(_.isString(str));
   return new BN(str, base);
-};
+}
 
-BN.fromBuffer = function(buf, opts) {
+BN.fromString = fromString;
+
+function fromBuffer(buf, opts) {
+  let localbuf;
   if (typeof opts !== 'undefined' && opts.endian === 'little') {
-    buf = reversebuf(buf);
+    localbuf = reversebuf(buf);
+  } else {
+    localbuf = buf;
   }
-  var hex = buf.toString('hex');
-  var bn = new BN(hex, 16);
+  const hex = localbuf.toString('hex');
+  const bn = new BN(hex, 16);
   return bn;
-};
+}
+BN.fromBuffer = fromBuffer;
 
 /**
  * Instantiate a BigNumber from a "signed magnitude buffer"
  * (a buffer where the most significant bit represents the sign (0 = positive, -1 = negative))
  */
-BN.fromSM = function(buf, opts) {
-  var ret;
+function fromSM(buf, opts) {
+  let ret;
+  let localbuf;
   if (buf.length === 0) {
-    return BN.fromBuffer(new Buffer([0]));
+    return BN.fromBuffer(Buffer.alloc([0]));
   }
 
-  var endian = 'big';
+  let endian = 'big';
   if (opts) {
-    endian = opts.endian;
+    endian = opts;
```

Comment:
This is most likely wrong. I assume that the intention was to use the [destructing assignment](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment).